api/client/run.go

```diff
@@ -36,6 +38,27 @@ func (cid *cidFile) Write(id string) error {
 	return nil
 }
 
+// Exit code 125 signals docker failure
+// Exit code 126 signals contained cmd found but cannot be invoked
+// Exit code 127 for contained cmd cannot be found
+func runError(err error) error {
+	// syscall.ENOENT.Error() and exec.ErrNotFound.Error() strings vary per client
+	// Windows: "The system cannot find the file specified" and "executable file not found in %PATH%"
+	// Linux: "no such file or directory" and "executable file not found in $PATH"
+	if strings.Contains(err.Error(), "system cannot find the file specified") ||
+		strings.Contains(err.Error(), "no such file or directory") ||
+		strings.Contains(err.Error(), "executable file not found") {
+		return Cli.StatusError{StatusCode: 127}
+	}
+	if strings.Contains(err.Error(), syscall.EACCES.Error()) {
+		return Cli.StatusError{StatusCode: 126}
+	}
+	if strings.Contains(err.Error(), "latest not found") {
+		return Cli.StatusError{StatusCode: 125}
```

Comment:
why don't these `StatusError` (125,126,127) contain a `Status` field set to `err.Error` (like the generic 125)?

now if you hit an error from 125 (latest not found), 126 or 127 the output is:
```
root@224662c1534f:/go/src/github.com/docker/docker# docker run busybox /etc
exec: "/etc": permission denied
```
while before this patch (and with `Status` field populated) output is:
```
root@224662c1534f:/go/src/github.com/docker/docker# docker run busybox /etc
exec: "/etc": permission denied
Error response from daemon: Cannot start container 62f29bd2a514c57fd0708b6a7dddf62008b1fc73a1b4797eb497e34f39b20b19: [8] System error: exec: "/etc": permission denied
```
is this change intented?